src/Conf.ml

```diff
@@ -376,6 +377,13 @@ module Formatting = struct
     C.choice ~names ~all ~doc ~section (fun conf x ->
         {conf with break_struct= Poly.(x = `Force)} )
 
+  let compact_expressions =
+    let default = false in
+    let doc = "Style for lists and arrays." in
+    let names = ["compact-expressions"] in
+    C.flag ~default ~names ~doc ~section (fun conf compact_expressions ->
+        {conf with compact_expressions} )
```

Comment:
What about handling this like `break-struct`, where the representation is a bool but the option accepts named alternates? For this option, I think using the same choices as `break-infix` would work well. That would also make it easier to give better docs.